From d293623f0fbb9a737c4543cf85cd1cc256b1a669 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Fri, 18 May 2018 15:29:43 +0800
Subject: [PATCH 3853/5242] MLK-18355-4: imx hdp: Get cec clk div from
 SW_CLK_H register

commit  e337ec701c45cf8e74cf85ee44817cf02ed5237f from
https://source.codeaurora.org/external/imx/linux-imx.git

Remove get core clock rate code,
get cec clk div from HDMI SW_CLK_H register.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdp.c |    6 ++++--
 drivers/mxc/hdp-cec/imx-hdp-cec.c |    7 -------
 drivers/mxc/hdp-cec/imx-hdp-cec.h |    1 -
 3 files changed, 4 insertions(+), 10 deletions(-)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.c b/drivers/gpu/drm/imx/hdp/imx-hdp.c
index 58eafaa..f8b3a79 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.c
@@ -77,12 +77,14 @@ static void imx_hdp_state_init(struct imx_hdp *hdp)
 #ifdef CONFIG_IMX_HDP_CEC
 static void imx_hdp_cec_init(struct imx_hdp *hdp)
 {
+	state_struct *state = &hdp->state;
 	struct imx_cec_dev *cec = &hdp->cec;
+	u32 clk_MHz;
 
 	memset(cec, 0, sizeof(struct imx_cec_dev));
 
-	if (hdp->clks.clk_core)
-		cec->clk_core = hdp->clks.clk_core;
+	CDN_API_GetClock(state, &clk_MHz);
+	cec->clk_div = clk_MHz * 10;
 	cec->dev = hdp->dev;
 	cec->mem = &hdp->mem;
 	cec->rw = hdp->rw;
diff --git a/drivers/mxc/hdp-cec/imx-hdp-cec.c b/drivers/mxc/hdp-cec/imx-hdp-cec.c
index 8d23c5a..e1aea1f 100644
--- a/drivers/mxc/hdp-cec/imx-hdp-cec.c
+++ b/drivers/mxc/hdp-cec/imx-hdp-cec.c
@@ -272,13 +272,6 @@ int imx_cec_register(struct imx_cec_dev *cec)
 	struct device *dev = cec->dev;
 	int ret;
 
-	/* Set CEC clock divider */
-	if (cec->clk_core)
-		cec->clk_div = clk_get_rate(cec->clk_core) / 100000;
-	else
-		/* Default HDMI core clock rate 133MHz */
-		cec->clk_div = 1330;
-
 	cec->adap = cec_allocate_adapter(&imx_cec_adap_ops, cec,
 					 CEC_NAME,
 					 CEC_CAP_PHYS_ADDR | CEC_CAP_LOG_ADDRS |
diff --git a/drivers/mxc/hdp-cec/imx-hdp-cec.h b/drivers/mxc/hdp-cec/imx-hdp-cec.h
index 4a7593a8..42a06b9 100644
--- a/drivers/mxc/hdp-cec/imx-hdp-cec.h
+++ b/drivers/mxc/hdp-cec/imx-hdp-cec.h
@@ -336,7 +336,6 @@ struct imx_cec_dev {
 	struct task_struct *cec_worker;
 
 	/* inited by HDP controller driver */
-	struct clk *clk_core;
 	struct hdp_mem *mem;
 	struct hdp_rw_func *rw;
 };
-- 
1.7.9.5

